core/src/toga/app.py

```diff
@@ -322,10 +315,7 @@ def __init__(
             self.icon = icon
 
         # Set up the document types and collection of documents being managed.
-        self._documents = DocumentSet(
-            self,
-            types=[] if document_types is None else document_types,
-        )
+        self._document_types = [] if document_types is None else document_types
```

Comment:
I get why this has been done, but I don't love it as an approach - it feels a bit too much like the properties of one particular medium making the ergonomics for every other medium materially worse. At this "proof of concept" stage I can live with it; but as a long term solution, I'd rather see a dummy `DocumentSet` implementation that is used when the platform doesn't support documents.